pkg/kubelet/eviction/eviction_manager.go

```diff
@@ -425,3 +444,122 @@ func (m *managerImpl) reclaimNodeLevelResources(resourceToReclaim v1.ResourceNam
 	}
 	return false
 }
+
+// localVolumeEviction checks the EmptyDir volume usage for each pod and determine whether it exceeds the specified limit and needs
+// to be evicted. It also checks every container in the pod, if the container overlay usage exceeds the limit, the pod will be evicted too.
+func (m *managerImpl) localVolumeEviction(pods []*v1.Pod) bool {
+	summary, err := m.summaryProvider.Get()
+	if err != nil {
+		glog.Errorf("Could not get summary provider")
+		return false
+	}
+
+	statsFunc := cachedStatsFunc(summary.Pods)
+	evict := false
+	for _, pod := range pods {
+		podStats, ok := statsFunc(pod)
+		if !ok {
+			continue
+		}
+
+		if m.emptyDirLimitEviction(podStats, pod) {
+			evict = true
+		}
+
+		if m.containerOverlayLimitEviction(podStats, pod) {
+			evict = true
+		}
+	}
+
+	return evict
+}
+
+func (m *managerImpl) localStorageEviction(pods []*v1.Pod) []*v1.Pod {
+	summary, err := m.summaryProvider.Get()
+	if err != nil {
+		glog.Errorf("Could not get summary provider")
+		return nil
+	}
+
+	statsFunc := cachedStatsFunc(summary.Pods)
+	evicted := []*v1.Pod{}
+	for _, pod := range pods {
+		podStats, ok := statsFunc(pod)
+		if !ok {
+			continue
+		}
+
+		if m.emptyDirLimitEviction(podStats, pod) {
+			evicted = append(evicted, pod)
```

Comment:
It's odd that the same pod might end up being appended multiple times. Is there any de-duping happening?